cmd/minishift/cmd/stop.go

```diff
@@ -61,12 +61,16 @@ func runStop(cmd *cobra.Command, args []string) {
 
 	fmt.Println("Stopping local OpenShift cluster...")
 
-	if !skipUnRegistration {
-		if _, err := cluster.UnRegister(api); err != nil {
+	if !skipUnRegistration && minishiftConfig.InstanceConfig.IsRegistered {
+		supportUnregistration, err := cluster.UnRegister(api)
```

Comment:
I think the problem is that we try to do two things here. Determine whether the VM supports un-registration as well as doing the actual un-registration.